numpy/core/_type_aliases.py

```diff
@@ -127,27 +106,23 @@ def _add_aliases():
         if name in ('longdouble', 'clongdouble') and myname in allTypes:
             continue
 
-        base_capitalize = english_capitalize(base)
-        if base == 'complex':
-            na_name = '%s%d' % (base_capitalize, bit//2)
-        elif base == 'bool':
-            na_name = base_capitalize
-        else:
-            na_name = "%s%d" % (base_capitalize, bit)
-
         allTypes[myname] = info.type
 
         # add mapping for both the bit name and the numarray name
         sctypeDict[myname] = info.type
-        sctypeDict[na_name] = info.type
 
         # add forward, reverse, and string mapping to numarray
-        sctypeNA[na_name] = info.type
-        sctypeNA[info.type] = na_name
-        sctypeNA[info.char] = na_name
-
         sctypeDict[char] = info.type
-        sctypeNA[char] = na_name
+
+    # Add deprecated numeric-style type aliases manually, at some point
+    # we may want to deprecate the lower case "bytes0" version as well.
+    for name in ["Bytes0", "Datetime64", "Str0", "Uint64"]:
+        if name == "Uint64" and "uint64" not in allTypes:
+            # Due to the integer priority, this was not always defined
+            continue
```

Comment:
This if is weird, but it failed tests and I think it is correct... if anyone has a 32bit linux or maybe windows to see that `Uint64` was never included in `np.typeDict` on old versions, that would be good.

Reply:
It now passes tests, is this comment outdated?

Reply:
No, I worked around it. But, I now realized on windows we probably had a stray `Uint32` and that will be missing now...

Reply:
np.typeDict has `UInt64` on 1.16, windows 32 bit:
```
>>> sys.version
'2.7.18 (v2.7.18:8d21aa21f2, Apr 20 2020, 13:19:08) [MSC v.1500 32 bit (Intel)]'
>>> np.typeDict['UInt64']
<type 'numpy.uint64'>
>>> np.__version__
1.16.6
>>>
```

Reply:
note the 'I' in 'UInt64'

Reply:
The capitalized `UInt` form has already been deprecated - it's just the weird `Uint` form that isn't.

It looks to me like I added this type accidentally as part of giving `[np.cint, np.int_, np.longlong]` unique names.  On numpy 1.11.0 (and python 2.7) the name isn't there at all:
```python
>>> numpy.__version__
'1.11.0'
>>> sorted(k for k in numpy.typeDict.keys() if isinstance(k, str) and k[0].isupper() and len(k) > 2)
['Bool', 'Complex128', 'Complex32', 'Complex64', 'Datetime64', 'Float128', 'Float16', 'Float32', 'Float64', 'Int16', 'Int32', 'Int64', 'Int8', 'Object0', 'String0', 'Timedelta64', 'UInt16', 'UInt32', 'UInt64', 'UInt8', 'Unicode0', 'Void0']
```

So I think we can just remove this special case?